File: backend/internal/domain/order.go
package domain

import (
	"time"
)

// ExternalOrder represents an order from Swiggy/Zomato
type ExternalOrder struct {
	ID              int64      `db:"id" json:"id"`
	Platform        string     `db:"platform" json:"platform"` // "swiggy" or "zomato"
	ExternalOrderID string     `db:"external_order_id" json:"externalOrderId"`
	OrderDate       time.Time  `db:"order_date" json:"orderDate"`
	CustomerName    *string    `db:"customer_name" json:"customerName,omitempty"`
	TotalAmount     float64    `db:"total_amount" json:"totalAmount"`
	Status          string     `db:"status" json:"status"`
	ItemsJSON       *string    `db:"items_json" json:"itemsJson,omitempty"`
	RawData         *string    `db:"raw_data" json:"rawData,omitempty"`
	NotifiedAt      *time.Time `db:"notified_at" json:"notifiedAt,omitempty"`
	CreatedAt       time.Time  `db:"created_at" json:"createdAt"`
}

// OrderItem represents an item in an order (parsed from ItemsJSON)
type OrderItem struct {
	Name     string  `json:"name"`
	Quantity int     `json:"quantity"`
	Price    float64 `json:"price"`
}

// CreateOrderRequest is the request body for order ingestion
type CreateOrderRequest struct {
	Platform        string  `json:"platform" validate:"required,oneof=swiggy zomato"`
	ExternalOrderID string  `json:"externalOrderId" validate:"required"`
	OrderDate       string  `json:"orderDate" validate:"required"`
	CustomerName    *string `json:"customerName"`
	TotalAmount     float64 `json:"totalAmount" validate:"required,gte=0"`
	Status          string  `json:"status"`
	ItemsJSON       *string `json:"itemsJson"`
	RawData         *string `json:"rawData"`
}

// OrderFilters for listing orders
type OrderFilters struct {
	Platform  string
	StartDate *time.Time
	EndDate   *time.Time
	Limit     int
	Offset    int
}

// OrderStats for analytics
type OrderStats struct {
	TotalOrders   int     `json:"totalOrders"`
	TotalRevenue  float64 `json:"totalRevenue"`
	SwiggyOrders  int     `json:"swiggyOrders"`
	SwiggyRevenue float64 `json:"swiggyRevenue"`
	ZomatoOrders  int     `json:"zomatoOrders"`
	ZomatoRevenue float64 `json:"zomatoRevenue"`
	StartDate     string  `json:"startDate"`
	EndDate       string  `json:"endDate"`
}

File: mobile/lib/models/order.dart
import 'dart:convert';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'order.freezed.dart';
part 'order.g.dart';

@freezed
class ExternalOrder with _$ExternalOrder {
  const ExternalOrder._();

  const factory ExternalOrder({
    required int id,
    required String platform,
    required String externalOrderId,
    required String orderDate,
    String? customerName,
    required double totalAmount,
    required String status,
    String? itemsJson,
    String? notifiedAt,
    required String createdAt,
  }) = _ExternalOrder;

  factory ExternalOrder.fromJson(Map<String, dynamic> json) =>
      _$ExternalOrderFromJson(json);

  List<OrderItem> get parsedItems {
    if (itemsJson == null || itemsJson!.isEmpty) return [];
    try {
      final list = jsonDecode(itemsJson!) as List;
      return list.map((e) => OrderItem.fromJson(e as Map<String, dynamic>)).toList();
    } catch (_) {
      return [];
    }
  }
}

@freezed
class OrderItem with _$OrderItem {
  const factory OrderItem({
    required String name,
    required int quantity,
    required double price,
  }) = _OrderItem;

  factory OrderItem.fromJson(Map<String, dynamic> json) =>
      _$OrderItemFromJson(json);
}

@freezed
class OrderStats with _$OrderStats {
  const factory OrderStats({
    required int totalOrders,
    required double totalRevenue,
    required Map<String, PlatformStats> platformBreakdown,
  }) = _OrderStats;

  factory OrderStats.fromJson(Map<String, dynamic> json) =>
      _$OrderStatsFromJson(json);
}

@freezed
class PlatformStats with _$PlatformStats {
  const factory PlatformStats({
    required int count,
    required double revenue,
  }) = _PlatformStats;

  factory PlatformStats.fromJson(Map<String, dynamic> json) =>
      _$PlatformStatsFromJson(json);
}
